activestorage/lib/active_storage/service/azure_storage_service.rb

```diff
@@ -117,8 +117,10 @@ def url_for_direct_upload(key, expires_in:, content_type:, content_length:, chec
       end
     end
 
-    def headers_for_direct_upload(key, content_type:, checksum:, **)
-      { "Content-Type" => content_type, "Content-MD5" => checksum, "x-ms-blob-type" => "BlockBlob" }
+    def headers_for_direct_upload(key, content_type:, checksum:, filename: nil, disposition: nil, **)
+      content_disposition = content_disposition_with(type: disposition, filename: filename) if filename
+
+      { "Content-Type" => content_type, "Content-MD5" => checksum, "x-ms-blob-content-disposition" => content_disposition, "x-ms-blob-type" => "BlockBlob" }
```

Comment:
This is unrelated, but are `Content-Type` and `Content-MD5` valid here? According to [the docs](https://docs.microsoft.com/en-us/rest/api/storageservices/set-blob-properties) it should be `x-ms-blob-content-type` and `x-ms-blob-content-md5`.

Reply:
I just wrote a test, I think it does work. Out of scope of this PR, but do you want me to PR a test for this?

Reply:
PR'd this in #36901.